Azure/sonic-snmpagent
# [201911][multi-asic][RFC1213]: Oper status of mgmt interface for multi-asic platform

[201911][multi-asic][RFC1213]: Retrieve oper status of management
interface from kernel for multi-asic platform.
In multi-asic platform there is no daemon in the host
namespace to update the oper status of management interface.
Instead of retrieving this information from state_db,
retrieve this from /sys/class/net/../operstate file.

Signed-off-by: Suvarna Meenakshi <sumeenak@microsoft.com>



- What I did
This PR is same as : https://github.com/Azure/sonic-snmpagent/pull/209; for 201911
Currently oper status of all management interface is retrieved from STATE DB.
Multi-asic platform does not have any daemon running on the host namespace to update oper status of management interface in STATE DB. This PR is to add a function to retrieve oper status of management interface from kernel file: "/sys/class/net/if_name/operstate".
- How I did it

Add a new function to retrieve oper status from /sys/class/net/if_name/operstate.
Invoke this function periodically to retrieve oper status of all management interfaces.
This is done periodically, instead of retrieving only on SNMP query, so that the time take during the SNMP query does not increase due to the file read operation.
Add unit-tests for mgmt interface.
- How to verify it
Run snmpwalk to get interface status, ensure that the interface status of management interface is correct, on single asic and multi-asic VS testbed.
With the changes - no impact of single asic platform.
**- Description for the changelog**
